S3Proxy/cache.go

```diff
@@ -5,45 +5,104 @@ import (
 	"time"
 )
 
-var s3Buckets = map[string]*s3BucketCacheItem{}
+var s3Buckets = map[string]*S3BucketCacheItem{}
+var s3Objects = map[string]*S3ObjectCacheItem{}
 
-type s3BucketCacheItem struct {
-	Name      string
-	Location  string
+type s3CacheItem struct {
```

Comment:
I can't see us using the CacheItem directly because all it contains is a timestamp and a TTL.

For that reason there is no reason to export it and if you don't export something it must be lowercase.